certification/internal/shell/podman.go

```diff
@@ -284,14 +284,18 @@ func (pe PodmanCLIEngine) Unshare(env map[string]string, command ...string) (*cl
 	return &cli.PodmanUnshareReport{Stdout: stdout.String(), Stderr: stderr.String()}, err
 }
 
-func (pe PodmanCLIEngine) UnshareWithCheck(check, image string, command ...string) (*cli.PodmanUnshareCheckReport, error) {
+func (pe PodmanCLIEngine) UnshareWithCheck(check, image string, mounted bool) (*cli.PodmanUnshareCheckReport, error) {
 	env := map[string]string{
 		"PATH":                 os.Getenv("PATH"),
 		"PREFLIGHT_EXEC_CHECK": check,
 		"PREFLIGHT_EXEC_IMAGE": image,
 	}
 
-	unshareReport, err := pe.Unshare(env, command...)
+	if mounted {
+		env["PREFLIGHT_EXEC_MOUNTED"] = fmt.Sprintf("%t", mounted)
+	}
```

Comment:
Why are we tracking the mounted state for this function? Is there logic somewhere that utilizes `PREFLIGHT_EXEC_MOUNTED` for some purpose?

I see it was added to the runtime.Config as well; trying to get a more complete picture but I'm missing it reviewing this.

Reply:
This particular check does not require a mounted image. In fact, it will fail if it has one. So, I've added the ability for a check to designate whether it needs to have a mounted image or not. The environment variable then allows the "check run" command to signify whether it should instantiate a MountedEngine, or just a CheckEngine.